AsyncDisplayKit/Details/ASChangeSetDataController.mm

```diff
@@ -98,6 +98,10 @@ - (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
     for (_ASHierarchyItemChange *change in [_changeSet itemChangesOfType:_ASHierarchyChangeTypeInsert]) {
       [super insertRowsAtIndexPaths:change.indexPaths withAnimationOptions:change.animationOptions];
     }
+    
+    for (_ASHierarchyItemChange *change in [_changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]) {
+      [super moveRowAtIndexPath:change.fromIndexPath toIndexPath:change.toIndexPath withAnimationOptions:change.animationOptions];
+    }
```

Comment:
The tricky bit here is, we need to grab the "move from" objects before we do any deleting/inserting, then delete them along with any other delete operations, then re-insert them along with any other insert operations.

According to the [rules of batch updating](https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/TableView_iPhone/ManageInsertDeleteRow/ManageInsertDeleteRow.html#//apple_ref/doc/uid/TP40007451-CH10-SW17), we interpret delete's and move-from's as "before the update" indexes and move-to's as "after the update" indexes.

For example, imagine we start with [A, B, C] and we receive changeset [ delete(0), move(1, 1) ]. The correct way to process this is:

```
// Gather moved items
id movedItem = items[1]; // [A, B, C]
// Deletes & move-froms, in descending order.
items.delete(1) // [ A, C ]
items.delete(0) // [ C ]
// Inserts & move-tos, in ascending order.
items.insert(1, movedItem) // [ C, B ]
```

If we process it like the code above, we get:

```
items.delete(0) // [ B, C ]
items.move(1, 1) // [ B, C ]
```

It's tricky! So ASDataController should not implement `moveItem:` but instead should support `insertItemWithNode:` or something similar I think.